### ISSUE TITLE
No hot-reloading support

### ISSUE THREAD

**Issue by tmcdos** [intlify/eslint-plugin-vue-i18n#9]
After you run `npm run dev` and start working on your Vue application, if you add a new localization string in the JSON locale file and then try to use it in your template - it is not catched by this plugin and it continues to produce warnings for the rule `vue/no-missing-keys` until you restart the Webpack-dev-server.

Even if I change the function `loadLocaleMessages` in file `/lib/utils/index.js` to watch the filesystem there is still some race condition and I have to save my Vue components twice before the plugin catches the changed JSON file:
```javascript
function loadLocaleMessages (pattern) {
  const files = glob.sync(pattern)
  return files.map((file,index) => {
    const path = resolve(process.cwd(), file)
    const filename = file.replace(/^.*(\\|\/|:)/, '')
    const messages = require(path)
    const result = { fullpath: path, path: file, filename, messages }

    // we only watch the existing files - so we can not catch the adding of new files to the directory
    fs.watch(path, (event, filename) => {
      if (filename) {
        delete require.cache[require.resolve(path)] // invalidate the cache
        result.messages = require(path) // does not handle deleting/renaming a file
      }
    })
    return result
  })
}
```

**Comment by tmcdos** [comment]
Okay, for the moment my workaround for this issue is to simply reload the messages every time (I could not find a way to stop watching the file(s) when ESLint is finished and thus it never returns to the command prompt):
```javascript
function loadLocaleMessages (pattern) {
  const files = glob.sync(pattern)
  return files.map(file => {
    const path = resolve(process.cwd(), file)
    const filename = file.replace(/^.*(\\|\/|:)/, '')
    delete require.cache[require.resolve(path)]
    const messages = require(path)
    return { fullpath: path, path: file, filename, messages }
  })
}

function getLocaleMessages (localeDirectory) {
  return loadLocaleMessages(localeDirectory)
} 
```

**Comment by tmcdos** [comment]
The issue seems to be caused by ESLint not re-linting Vue templates if the locale file(s) is modified - you have to first create the missing keys in the JSON file and save it, then make any trivial change to the corresponding single-file component in order to trigger re-linting and thus re-read the new JSON file.

**Comment by kazupon** [comment]
Thanks a lot feedback!

**Comment by johan-ludvigsson-stratsys** [comment]
Any update on this? We just ran into this problem as well.
